test/suites/drift/suite_test.go

```diff
@@ -719,6 +719,48 @@ var _ = Describe("Drift", func() {
 		env.EventuallyExpectNotFound(pod, node)
 		env.EventuallyExpectHealthyPodCount(selector, numPods)
 	})
+	It("should update nodepool-hash and nodepool-hash-version and re-propagates to the NodeClaim", func() {
+		env.ExpectCreated(dep, nodeClass, nodePool)
+		env.EventuallyExpectHealthyPodCount(selector, numPods)
+		nodeClaim := env.EventuallyExpectCreatedNodeClaimCount("==", 1)[0]
+		err := env.Client.Get(env.Context, client.ObjectKeyFromObject(nodePool), nodePool)
+		Expect(err).ToNot(HaveOccurred())
+		expectedHash := nodePool.Hash()
+
+		By("expect nodepool and nodeclaim to contain karpenter.sh/nodepool-hash annotation")
+		Eventually(func(g Gomega) {
+			g.Expect(nodePool.Annotations).To(HaveKeyWithValue(corev1beta1.NodePoolHashAnnotationKey, expectedHash))
+			g.Expect(nodePool.Annotations).To(HaveKeyWithValue(corev1beta1.NodePoolHashVersionAnnotationKey, corev1beta1.NodePoolHashVersion))
+			g.Expect(nodeClaim.Annotations).To(HaveKeyWithValue(corev1beta1.NodePoolHashAnnotationKey, expectedHash))
+			g.Expect(nodeClaim.Annotations).To(HaveKeyWithValue(corev1beta1.NodePoolHashVersionAnnotationKey, corev1beta1.NodePoolHashVersion))
+		}).WithTimeout(30 * time.Second).Should(Succeed())
+
+		nodePool.Annotations = lo.Assign(nodePool.Annotations, map[string]string{
+			corev1beta1.NodePoolHashAnnotationKey:        "test-version-1",
+			corev1beta1.NodePoolHashVersionAnnotationKey: "test-version-1",
+		})
+		nodeClaim.Annotations = lo.Assign(nodePool.Annotations, map[string]string{
+			corev1beta1.NodePoolHashAnnotationKey:        "test-version-2",
+			corev1beta1.NodePoolHashVersionAnnotationKey: "test-version-2",
+		})
+
+		// NodeClaim will need to be updated first, as the hash controller will not be reconciled on NodeClaim updates
+		env.ExpectUpdated(nodeClaim, nodePool)
```

Comment:
So, nothing changed on the NodeClaim or NodePool that would cause anything to drift anyways. I think to actually ensure what you want to ensure here, you need to change some values on the NodePool so that you can put yourself into a situation where you would normally expect some drift to occur, but none occurs because the drift hash version annotation didn't match